packages/eas-cli/src/commands/update/index.ts

```diff
@@ -586,15 +610,43 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return [transformRuntimeVersions(exp, platforms), exp];
+  } catch (error: any) {
+    if (nonInteractive) {
+      Errors.error(error, { exit: 1 });
+    }
+
+    Log.failed(error.message);
+
+    const runConfig = await selectAsync(
+      `Do you want us to run automatic ${chalk.bold(
+        'app.json'
+      )} configuration for EAS Update for you?`,
+      [
+        { title: 'Yes', value: true },
+        {
+          title: 'No, I will set the runtime version manually (EAS CLI exits)',
+          value: false,
+        },
+      ]
+    );
+
+    if (!runConfig) {
+      Errors.exit(1);
```

Comment:
ditto about throwing an exception rather than using `Errors.exit` (can grep for both to see the relative usage of each, and tend to prefer the more common pattern)

Reply:
This is a conscious decision to exit without an error here, since the user have chose this option. Throwing an another error here will not add any value or information, can only confuse user.

https://github.com/expo/eas-cli/blob/123615212058bf841dec00edcc16da284c27e148/packages/eas-cli/src/commands/build/index.ts#L300-L307

Reply:
> This is a conscious decision to exit without an error here,

`Errors.exit(1)` is not exiting without an error.

>  can only confuse user.

I don't see how, if you throw sth along the lines of "Aborting ..."

> Also this pattern has been taken from the codebase...

Similar patterns are not really a good argument, sth that is fine there might not be fine here. This "pattern" is used in very different contexts:
- In build, it's used in code that is run at the very start of the process, as part of the input/config validation. That function is handling deprecated config for something very specific.
- In update code that exit is run in the util function that someone might want to extract to common code in the future.

With that said I think leaving exit might be fine, but in this case, I think it's a bit borderline.

Reply:
Under the hood all `Errors.exit` does is throw an error: https://github.com/oclif/core/blob/main/src/errors/index.ts#L18

I don't feel too strongly here, but now that I grep for similar cases I see a mix of `throw`, `Errors.exit` and `process.exit`. `process.exit` is definitely incorrect since it will cause the tear down code of oclif commands not to run (see EasCommand.finally). `Errors.exit` and `throw` are equivalent, so it's just a matter of style.

I'd say this is fine here, but really what we should do in a follow-up is a brief audit of all cases followed by a potential lint rule to ensure that this remains consistent going forward across the codebase. That way there's no attention cost to this anymore.

For this PR, it's fine to keep it as you have it.